test-app/tests/integration/components/flash-message-test.js

```diff
@@ -18,23 +19,29 @@ module('Integration | Component | flash message', function (hooks) {
   setupRenderingTest(hooks);
 
   test('it renders a flash message', async function (assert) {
-    this.set('flash', FlashMessage.create({ message: 'hi', sticky: true }));
+    const flash = new FlashMessage({ message: 'hi', sticky: true });
```

Comment:
does the FlashMessage need `associateDestroyableChild` when it's created?

Reply:
Yes, good idea. I've implemented this in `FlashMessagesService` when the messages are created